src/main/java/uk/gov/hmcts/reform/prl/services/tab/alltabs/AllTabServiceImpl.java

```diff
@@ -70,19 +78,52 @@ public void updateAllTabsIncludingConfTab(CaseData caseData) {
         combinedFieldsMap.putAll(confidentialDetails);
 
         if (caseData.getDateSubmitted() != null) {
-            combinedFieldsMap.put("dateSubmitted", caseData.getDateSubmitted());
+            combinedFieldsMap.put(DATE_SUBMITTED_FIELD, caseData.getDateSubmitted());
         }
         if (caseData.getCourtName() != null) {
-            combinedFieldsMap.put("courtName", caseData.getCourtName());
+            combinedFieldsMap.put(COURT_NAME_FIELD, caseData.getCourtName());
         }
         if (caseData.getCourtId() != null) {
-            combinedFieldsMap.put("courtId", caseData.getCourtId());
+            combinedFieldsMap.put(COURT_ID_FIELD, caseData.getCourtId());
         }
-        getDocumentsMap(caseData,combinedFieldsMap);
+        getDocumentsMap(caseData, combinedFieldsMap);
         // Calling event to refresh the page.
         refreshCcdUsingEvent(caseData, combinedFieldsMap);
     }
 
+    public void updateAllTabsIncludingConfTabRefactored(String authorisation,
+                                                        String caseId,
+                                                        StartEventResponse startEventResponse,
+                                                        EventRequestData allTabsUpdateEventRequestData,
+                                                        CaseData caseData) {
+        Map<String, Object> confidentialDetails = confidentialityTabService.updateConfidentialityDetails(caseData);
+        Map<String, Object> combinedFieldsMap = getCombinedMap(caseData);
+        combinedFieldsMap.putAll(confidentialDetails);
+
+        if (caseData.getDateSubmitted() != null) {
+            combinedFieldsMap.put(DATE_SUBMITTED_FIELD, caseData.getDateSubmitted());
+        }
+        if (caseData.getCourtName() != null) {
+            combinedFieldsMap.put(COURT_NAME_FIELD, caseData.getCourtName());
+        }
+        if (caseData.getCourtId() != null) {
+            combinedFieldsMap.put(COURT_ID_FIELD, caseData.getCourtId());
+        }
+        getDocumentsMap(caseData, combinedFieldsMap);
+        refreshCcdUsingEvent(caseData, combinedFieldsMap);
```

Comment:
We are submitting the event just one line below. how this line 113 is significant?